File: examples/quickstart.py
"""
Example: Headless Threads keyword search, discussion tree scraping & auto-reply.
"""
from threadsflow import ThreadsAPI, ThreadsSession

def main():
    # 1. Real-time keyword discovery (zero proxy cost, sub-80ms)
    threads = ThreadsAPI(device_preset="threads_ios")
    print("Searching public discussions for keyword 'ai agents'...")
    posts = threads.search_posts(query="ai agents", limit=5)
    for post in posts:
        print(f"[@{post.author.username}] ({post.like_count} likes): {post.caption[:80]}...")
        print(f"Post ID: {post.id} | Code: {post.code}")

    # 2. Authenticated replies (using shared Meta session)
    # session = ThreadsSession.load_from_file("session_threads.json")
    # auth_client = ThreadsAPI(session=session)
    # auth_client.reply(parent_post_id="3141592653589793238", text="Great perspective!")

if __name__ == "__main__":
    main()

File: ts/examples/quickstart.ts
/**
 * Quickstart: Headless Threads keyword discovery and discussion tree inspection.
 */
import { ThreadsAPI, ThreadsSession } from "../src";

async function main() {
  // 1. Unauthenticated or session-based client
  const threads = new ThreadsAPI({
    devicePreset: "threads_ios",
  });

  console.log("Searching public discussions for 'ai agents'...");
  const posts = await threads.searchPosts("ai agents", 5);

  for (const post of posts) {
    console.log(`[@${post.author.username}] (${post.likeCount} likes): ${post.caption.slice(0, 80)}...`);
    console.log(`Post ID: ${post.id} | Code: ${post.code}`);
  }

  // 2. Fetch full discussion tree
  if (posts.length > 0) {
    const targetId = posts[0].id;
    console.log(`\nFetching discussion tree for post ${targetId}...`);
    const tree = await threads.getThread(targetId);
    console.log(`Root OP by @${tree.rootPost.author.username}: ${tree.rootPost.caption}`);
    console.log(`Found ${tree.replies.length} replies.`);
  }

  // 3. Authenticated session reply example
  // const session = ThreadsSession.loadFromFile("session_threads.json");
  // const authClient = new ThreadsAPI({ session });
  // await authClient.reply("3141592653589793238", "Great perspective on headless architecture!");
}

main().catch(console.error);
